src/components/graphQLConnect.ts

```diff
@@ -186,6 +186,23 @@ class GraphQLdb {
     )
     return data?.application.template.templatePermissions.nodes || null
   }
+
+  public getFilePaths = async (batchSize: number, offset: number) => {
+    const data = await this.gqlQuery(
+      `
+      query getFilePaths($first:Int!, $offset:Int!) {
+        files(first: $first, offset: $offset) {
+          nodes {
+            filePath
+            id
+          }
+        }
+      }
+      `,
+      { first: batchSize, offset }
+    )
+    return data?.files?.nodes || []
+  }
```

Comment:
GraphQL query is easier for this, as we can use pagination to grab them in batches -- much harder to do with SQL without using a CURSOR.